### ISSUE TITLE
Select the whole scrollback buffer keybinding not working

### ISSUE THREAD

**Issue by blueray453** [alacritty/alacritty#6204]
I am using the following keybinding to select everything from the scrollback buffer.

```
  - { key: A,  mods: Control,              action: ToggleViMode                       }
  - { key: A,  mods: Control, mode: Vi,    action: ScrollToTop                     }
  - { key: A,  mods: Control, mode: Vi,    action: ToggleNormalSelection   }
  - { key: A,  mods: Control, mode: Vi,    action: ScrollToBottom                }
  - { key: A,  mods: Control, mode: Vi,    action: ToggleViMode                 }
```
My target is, I turn on vi mode, go to top. start selecting, go to bottom (selecting everything in between), get out of vi mode.

But i am not even understanding what this keybinding is doing?

Please help me come up with a keybinding that will select  the whole scrollback buffer

### System

OS: Linux Mint 20.3
Version: alacritty 0.10.1 (c3851b7)
Linux/BSD: X11

**Comment by chrisduerr** [comment]
Why did you define a keybinding when you don't even know what it does?

I don't think it's currently possible to do what you want with a single key binding.

**Comment by blueray453** [comment]
If I set:

```
  - { key: A,  mods: Control, mode: Vi,  action: ScrollToTop            }
  - { key: A,  mods: Control, mode: Vi,    action: ToggleNormalSelection   }
  - { key: A,  mods: Control, mode: Vi,    action: ScrollToBottom                }
```


enter vi mode and then run Ctrl+A then it select the whole scrollback buffer.

The thing is I want to enter and exit vi mode with ctrl+A as well. 

What i mean is:

```
  - { key: A,  mods: Control,                    action: ENTER_VI_MODE         }
  - { key: A,  mods: Control, mode: Vi,    action: ScrollToTop                     }
  - { key: A,  mods: Control, mode: Vi,    action: ToggleNormalSelection   }
  - { key: A,  mods: Control, mode: Vi,    action: ScrollToBottom                }
  - { key: A,  mods: Control,                    action: EXIT_VI_MODE              }  
```

**Comment by chrisduerr** [comment]
Yeah, that doesn't work. IIRC there should be another issue open somewhere about modes being evaluated ahead of time too.

**Comment by blueray453** [comment]
While you are at it please add vscode mode as well. What i mean is, in tmux, i do something like

```
# Ctrl+e to view the output buffer in vscode
bind-key -T root C-e run-shell "tmux capture-pane -pS -50000 > /tmp/tmux-code.txt; codium /tmp/tmux-code.txt"
```
What I am doing here is saving the whole scrollback buffer in `/tmp/tmux-code.txt` and then opening it with vscode (in this case codium). So, i can search, copy etc everything there.

If I could copy the whole scrollback buffer with one key press, my next step would be to run shell commands as action.

What i mean in:

  - { key: E,  mods: Control,                    action: ENTER_VI_MODE         }
  - { key: E,  mods: Control, mode: Vi,    action: ScrollToTop                     }
  - { key: E,  mods: Control, mode: Vi,    action: ToggleNormalSelection   }
  - { key: E,  mods: Control, mode: Vi,    action: ScrollToBottom                }
  - { key: E,  mods: Control, mode: Vi,    action: Copy              }  
  - { key: E,  mods: Control, mode: Vi,    action: 'xsel --clipboard > /tmp/tmux-code.txt '          }  
  - { key: E,  mods: Control, mode: Vi,    action: 'codium /tmp/tmux-code.txt '          }  

That way, i would not need to use the search or vi mode at all.

**Comment by blueray453** [comment]
I can already do it but take few more steps.

1. Ctrl+Shift+Space
2. Ctrl+A
3. Ctrl+C
4. Open Vscode
5. Paste

**Comment by blueray453** [comment]
In case anyone is interested in my current solution, I am using:

```
  - { key: E,  mods: Control, mode: Vi,    action: ScrollToTop }
  - { key: E,  mods: Control, mode: Vi,    action: ToggleNormalSelection }
  - { key: E,  mods: Control, mode: Vi,    action: ScrollToBottom }
  - { key: E,  mods: Control, mode: Vi,    action: Copy }
  - { key: E,  mods: Control, mode: Vi,    action: ClearSelection  }
  - { key: E,  mods: Control, mode: Vi,    action: ToggleViMode }
```

I am also using a zsh widget like:

```
_open_clipboard_in_file() {
  xsel -b > /tmp/alacritty-codium.txt
  codium /tmp/alacritty-codium.txt
}

zle -N _open_clipboard_in_file
bindkey '^E' _open_clipboard_in_file # Copy text
```

So, now when i press `Ctrl+Shift+Space > Ctrl+E > Ctrl+E`, it open the whole scrollback buffer in codium (you can use your favorite code editor).

> Yeah, that doesn't work. IIRC there should be another issue open somewhere about modes being evaluated ahead of time too.

As far I understood, due to lack of **modes not being evaluated ahead of time**, we have to use three keybindings instead of one.

In case they decide to add this feature (evaluated modes ahead of time), the keybinding will probably look like:

```
  - { key: E,  mods: Control,  action: ToggleViMode }
  - { key: E,  mods: Control, mode: Vi,    action: ScrollToTop }
  - { key: E,  mods: Control, mode: Vi,    action: ToggleNormalSelection }
  - { key: E,  mods: Control, mode: Vi,    action: ScrollToBottom }
  - { key: E,  mods: Control, mode: Vi,    action: Copy }
  - { key: E,  mods: Control, mode: Vi,    action: ClearSelection  }
  - { key: E,  mods: Control, mode: Vi,    action: ToggleViMode }
  - { key: E,  mods: Control,    action: ReceiveChar }
```
Here just pressing `Ctrl+E` will open the whole scrollback buffer in your favorite editor.
